### PR TITLE
Feature/register validations

### PR DESCRIPTION
**Pull request by jimmy404**

### CODE REVIEW

File: src/Components/Register/registerValidations.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    if (!value && value !== 0){
+      return msgErrorInputFieldRequired;
+    }
+    if (!value.match(emailRegexp)){
+      return msgErrorInputEmail;
+    }
+    return null;
+  },
+  userName: ({ value }) => {
+    const userNameRegexp = /^[a-z\d_]{6,20}$/i;
```
Comment (naueru) [inline src/Components/Register/registerValidations.js:46]: Try setting this min/max values from the config or just remove it from the regex

File: src/Components/Register/registerValidations.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+msgErrorInputUserName      = msgErrorInput.userName,
+msgErrorInputPassword      = msgErrorInput.password,
+msgErrorInputAlterEgo      = msgErrorInput.alterEgo,
+msgErrorInputIsPwdMatched  = msgErrorInput.isPwdMatched,
+msgErrorInputFieldRequired = msgErrorInput.fieldRequired;
+
+const registerValidator = {
+  email: ({ value }) => {
+    //eslint-disable-next-line
+    const emailRegexp =  /^\w+([\.\+\-]?\w+)*@([\.-]?\w+)*(\.\w{2,4})+$/;
```
Comment (naueru) [inline src/Components/Register/registerValidations.js:33]: Same as below

File: src/CritCore/Translations/Translations.js [hunk 3]
```diff
@@ -244,7 +244,15 @@ let translations = {
       alterEgoTooltip: 'Nombre visible / Alter ego',
       icon: 'Elija un avatar',
       submit: 'Crear',
-      selectPlaceholder: 'Elija un ícono'
+      selectPlaceholder: 'Elija un ícono',
+      errors: {
+        email: 'Email inválido',
+        userName:'UserName inválido',
+        password:'Caracteres: Min 4 | Max 30',
```
Comment (naueru) [inline src/CritCore/Translations/Translations.js]: Please set this translation as dependant of config. Check other similar translations

File: src/CritCore/Translations/Translations.js [hunk 4]
```diff
@@ -502,7 +510,15 @@ let translations = {
       alterEgoTooltip: 'Alter Ego / Visual name',
       icon: 'Choose an avatar',
       submit: 'Create',
-      selectPlaceholder: 'Select an icon'
+      selectPlaceholder: 'Select an icon',
+      errors: {
+        email: 'Invalid email',
+        userName:'Invalid username',
+        password:'Characters: Min 4 | Max 30',
```
Comment (naueru) [inline src/CritCore/Translations/Translations.js]: Same as his spanish counterpart

**Review by naueru — CHANGES_REQUESTED** [review]

File: src/Components/Register/registerValidations.js (hunk 2 above)
Comment (jimmy404) [reply src/Components/Register/registerValidations.js:33]: I couldn't remove the values ​​by checking them in as variables.
However, we were talking about leaving the minimum / maximum for the email, in order to validate up to 4 characters after the period (.1234)

File: src/Components/Register/registerValidations.js (hunk 1 above)
Comment (jimmy404) [reply src/Components/Register/registerValidations.js:46]: I Can't remove min-max without breaking user validation scheme.

File: src/CritCore/Translations/Translations.js (hunk 3 above)
Comment (jimmy404) [reply src/CritCore/Translations/Translations.js]: Done.

File: src/CritCore/Translations/Translations.js (hunk 4 above)
Comment (jimmy404) [reply src/CritCore/Translations/Translations.js]: Done.

File: src/CritCore/Translations/Translations.js [hunk 5]
```diff
@@ -244,7 +244,17 @@ let translations = {
       alterEgoTooltip: 'Nombre visible / Alter ego',
       icon: 'Elija un avatar',
       submit: 'Crear',
-      selectPlaceholder: 'Elija un ícono'
+      selectPlaceholder: 'Elija un ícono',
+      errors: {
+        email: 'Email inválido',
+        emailMinMaxLength: 'Caracteres: Min 6 | Max 60',
```
Comment (naueru) [inline src/CritCore/Translations/Translations.js]: Set this numbers from the config, that way the translations will adjust dynamically. The same for username lenght and their respective english equivalents.

File: src/CritCore/Translations/Translations.js [hunk 6]
```diff
@@ -244,7 +244,17 @@ let translations = {
...
+        userName:'UserName inválido',
```
Comment (naueru) [inline src/CritCore/Translations/Translations.js]: This should be completely in spanish.

File: src/CritCore/Translations/Translations.js (hunk 5 above)
Comment (jimmy404) [reply src/CritCore/Translations/Translations.js]: Done.

File: src/CritCore/Translations/Translations.js (hunk 6 above)
Comment (jimmy404) [reply src/CritCore/Translations/Translations.js]: Done.

**Review by naueru — APPROVED** [review]

**Pull request merged**